app/src/ui/add-repository/add-existing-repository.tsx

```diff
@@ -81,18 +103,39 @@ export class AddExistingRepository extends React.Component<IAddExistingRepositor
     this.checkIfPathIsRepository(path)
   }
 
+  private pathExists(directory: string): Promise<boolean> {
+    return new Promise<boolean>((resolve, reject) => {
+      try {
+        FS.stat(directory, (err, exists) => {
+          if (err) {
+            resolve(false)
+          }
+
+          resolve(true)
+        })
+      } catch (e) {
+        resolve(false)
+      }
+    })
+  }
+
   private async checkIfPathIsRepository(path: string) {
-    this.setState({ path, isGitRepository: null })
+    this.setState({ path })
```

Comment:
I believe we still need to set `isGitRepository` to null here, right? To indicate that we're currently in the process of evaluating whether it exists or not.